core/base/discreteGradient/DiscreteGradient.cpp

```diff
@@ -24,12 +24,31 @@ void DiscreteGradient::initMemory(const AbstractTriangulation &triangulation) {
   }
 
   // clear & init gradient memory
-  for(int i = 0; i < dimensionality_; ++i) {
-    (*gradient_)[2 * i].clear();
-    (*gradient_)[2 * i].resize(numberOfCells[i], -1);
-    (*gradient_)[2 * i + 1].clear();
-    (*gradient_)[2 * i + 1].resize(numberOfCells[i + 1], -1);
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp parallel master num_threads(threadNumber_)
+  {
+#endif
+    for(int i = 0; i < dimensionality_; ++i) {
+#ifdef TTK_ENABLE_OPENMP
+#pragma omp task
+      {
+#endif
```

Comment:
I believe you can move this `#endif` before the opening bracket `{`.
then, you won't need the `#ifdef` for the corresponding closing bracket l. 39. (it's OK to have opening and closing brackets)